docs/plugins/development/creating-text-analysis-plugins.asciidoc

```diff
@@ -0,0 +1,130 @@
+[[creating-text-analysis-plugins]]
+=== Creating text analysis plugins
+
+Text analysis plugins provide {es} with custom {ref}/analysis.html[Lucene 
+analyzers, token filters, character filters, and tokenizers]. 
+
+[discrete]
+==== Plugin file structure
```

Comment:
Could this section be swapped with the next? Describing the API itself seems more important than how plugins are packaged.